impeller/renderer/backend/vulkan/command_pool_vk.h

```diff
@@ -5,36 +5,80 @@
 #pragma once
 
 #include <memory>
-#include <set>
 #include <thread>
 
 #include "flutter/fml/macros.h"
 #include "impeller/base/thread.h"
+#include "impeller/renderer/backend/vulkan/context_vk.h"
```

Comment:
The reason we generally do forward declarations is to avoid circular dependencies in header files (and sometimes to reduce compilation time but that is marginal). As long as it compiles that should be okay, but you might hit cases where you can't remove a forward reference.

Reply:
A good to know. I think these are all in the same source set but I'll have CI double check me :)